### ISSUE TITLE
Netatalk 3.2.x

### ISSUE THREAD

**Issue by andylemin** [Netatalk/netatalk#1336]
**Describe the bug**
Cannot run both Netatalk 3.2.x and Samba 4.19 on FreeBSD 14.1, as Netatalk requires talloc is present, and Samba requires it is absent.

**To Reproduce**
`pkg install samba419` automatically removes talloc (breaking Netatalk).

`pkg install talloc` automatically removes samba..

**Expected behavior**
Samba and Netatalk can coexist on the same server.

**Environment**
 - Server OS: FreeBSD 14.1
 - Netatalk Version 3.2.4

**Comment by andylemin** [comment]
Conflict message that is causing one or the other to be removed;
'talloc-2.3.4_1 conflicts with samba419-4.19.7 on /usr/local/lib/python3.11/site-packages/talloc.cpython-311.so'

I understand Netatalk recently went through a cleanup effort regarding talloc, seems this might have been the reason why it was the way it was.
Might be tricky to find an ideal balance/compromise.

Mixed client environments (windows and mac) are still common for many businesses.

**Comment by rdmark** [comment]
This is quite unfortunate, and IMHO the proper solution would to be to convince the FreeBSD port maintainer to repackage the samba port to use the system talloc library instead of the bundled code. This is what Debian is doing, for instance, since they have a strict policy against "vendored" libraries to avoid this kind of dependency conflict. (As a side note, the Samba project maintains talloc so I don't blame them for wanting to bundle it.)

What the netatalk3 port maintainer has done to work around this issue is to disable Spotlight functionality. Would this be an acceptable workaround for you too, or do you rely strongly on Spotlight in your deployment?

**Comment by andylemin** [comment]
I agree having the option to not bundle talloc in Samba is more consistent with best practices, especially when it means deploying .so's into common locations.

And Spotlight is definitely desirable, especially for large shares.

Maybe you could give both Samba and FreeBSD teams a nudge and see what they think? Ideally it should be changed on Samba/upstream side as this is probably not limited to FreeBSD.

**Comment by andylemin** [comment]
Makes sense for the main talloc libraries to be privately linked, otherwise it would be a bigger issue.

You can do the same for python libraries as well, so I wonder why that one is different. Sounds like a simple omission maybe.

Sure, I'll try and send a message as well this week 🙂

**Comment by rdmark** [comment]
A week later, and the samba419 port maintainer hasn't responded yet to a direct email, unfortunately...

I wonder if there's another forum for getting traction for this question?

**Comment by andylemin** [comment]
Hi,I also emailed him, and he’s not replied to me either sadly.I’m not on any FreeBSD mailing lists, but I’m sure there are some. If I find time between day-job, I’ll join one and re-raise the topic. Thanks for the nudgeOn 4 Aug 2024, at 16:41, Daniel Markstedt ***@***.***> wrote:﻿
A week later, and the samba419 port maintainer hasn't responded yet to a direct email, unfortunately...
I wonder if there's another forum for getting traction for this question?
